R/DataDiscoveryRetrieval.R

```diff
@@ -169,13 +174,15 @@
 #'   sampleMedia = c("Water", "water"),
 #'   statecode = "CT", # consider downloading only 1 state at a time
 #'   providers = "NWIS",
-#'   applyautoclean = FALSE
+#'   applyautoclean = FALSE,
+#'   ask = FALSE
 #' )
 #'
 #' # query by country code (e.g. Canada, countrycode = "CA")
 #' tada11 <- TADA_DataRetrieval(
 #'   startDate = "2015-01-01",
-#'   countrycode = "CA"
+#'   countrycode = "CA",
+#'   ask = FALSE
 #' )
 #' }
 #'
```

Comment:
Suggestion, add in a query by tribal area:

```suggestion
#' # query by Tribal Area Type
#' tada11 <- TADA_DataRetrieval(
#'     startDate = "2015-01-01",
#'     tribal_area_type = "American Indian Reservations",
#'     ask = TRUE)
```
But note, this fails at the quiet_whatWQPdata step:

![Screenshot 2025-01-23 at 2 14 03 PM](https://github.com/user-attachments/assets/e74b1511-317e-466c-bfda-b2ea47837b15)

I suspect this is an extent issue, since the bounds of this sf multipolygon are enormous. If I add the argument `tribe_name_parcel = "Pueblo of Sandia, New Mexico"`, it works fine. I wonder if an additional caveat can be added to the messaging of that error to indicate that the extent of the bounds may be too large? 

Also probably need a conditional that if "American Indian Reservations" then must specify at least one tribe name parcel?

Reply:
Additional query attempted:  
```
tada11 <- TADA_DataRetrieval(
     startDate = "2015-01-01",
     tribal_area_type = "American Indian Reservations", tribe_name_parcel = 	
         c("Pueblo of Sandia, New Mexico", 
           "Ysleta del Sur Pueblo", "Robinson Rancheria", "	
 Santee Sioux Nation, Nebraska"),
     ask = TRUE)
```

It was a pretty long wait to get to the prompt for download (maybe 1-2 min?) So I wonder if a conditional message at the top when the query of the sf area for tribal lands would be good to make sure folks don't think the function is just hanging/broken?

Reply:
@steeleb I added a conditional to require a name/parcel_no if a type is provided (tbh thought this was already in there). How do you feel about that as a solution?

Also a message before the slow step where the available data query happens.

Reply:
If other types cause the same issue, then absolutely, otherwise, it could probably be limited to checking for area/parcel if specific types (e.g. "American Indian Reservations"). I'd trust your gut call here, though!